src_cpp/opennnBridge/choose_activation_function.h

```diff
@@ -0,0 +1,124 @@
+             || ( (neural_network->get_trainable_layers_pointers()(i))->get_type_string() == "Convolutional" ))
+         {
+            if(activations_functions(i) == E_AF_Threshold){
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::Threshold);
+            }
+
+            if(activations_functions(i) == E_AF_SymmetricThreshold){
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::SymmetricThreshold);
+            }
+
+            if(activations_functions(i) == E_AF_Logistic){ 
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::Logistic);
+            }
+
+            if(activations_functions(i) == E_AF_HyperbolicTangent){
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::HyperbolicTangent);
+            }
+
+            if(activations_functions(i) == E_AF_Linear){
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::Linear);
+            }
+
+            if(activations_functions(i) == E_AF_RectifiedLinear){
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::RectifiedLinear);
+            }
+
+            if(activations_functions(i) == E_AF_ExponentialLinear){
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::ExponentialLinear);
+            }
+
+            if(activations_functions(i) == E_AF_ScaledExponentialLinear){
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::ScaledExponentialLinear);
+            }
+
+            if(activations_functions(i) == E_AF_SoftPlus){
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::SoftPlus);
+            }
+ 
+            if(activations_functions(i) == E_AF_SoftSign){
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::SoftSign);
+            }
+
+            if(activations_functions(i) == E_AF_HardSigmoid){
+                  dynamic_cast<PerceptronLayer*>(neural_network->get_trainable_layers_pointers()(i))->set_activation_function(PerceptronLayer::HardSigmoid);
+            }
+         }
+
+
+         //Probabilistic
+         if((neural_network->get_trainable_layers_pointers()(i))->get_type_string() == "Probabilistic")
```

Comment:
Comparing strings is inefficient (compare each letter to another).
Use get_type instead and compare it with enum number from Definition at line 61 of file layer.h (in opennn).
@evgenyan95

Reply:
don't work